### PR TITLE
ui,sql: add Vectorized info to statements page

### PR DESCRIPTION
**Pull request by yuzefovich**
This commit adds a box to Statements page that displays whether the
vectorized execution engine is used.

Release note (admin ui change): Statements page now contains information
about which execution engine is used (vectorized or not).

### CODE REVIEW

**Comment by awoods187** [comment]
Thanks for jumping on this! Do you have a screenshot of what this will look like?

**Comment by yuzefovich** [comment]
Yes, I do (of the relevant section). I made the screenshot yesterday but forgot to add it here. Let me know if you're interested in the shot of the whole screen.
Screen Shot 2020-06-15 at 9 46 06 PM

**Comment by awoods187** [comment]
Thanks for sharing! What does 1 of 2 mean?

**Comment by yuzefovich** [comment]
It is the same as for `Distributed execution`: we represent that information as a "fraction", so 1 of 2 means that out of two times the query was executed, it ran once via the vectorized engine and once via the old engine. In edge cases we show the words: "no" if we have 0 out of any number of runs, and "yes" if the numerator of the "fraction" equals the denominator.

**Comment by awoods187** [comment]
Gotcha. I predict that it will be confusing to the user why the query fingerprint is sometimes run in vectorized and sometime not. 

In fact, why would that be the case? Is this an indication that we shouldn't actually be grouping these two queries under the same fingerprint?

**Comment by yuzefovich** [comment]
> I predict that it will be confusing to the user why the query fingerprint is sometimes run in vectorized and sometime not.

Yeah, it might be confusing, but we use the exactly same representation for "Distributed execution" (as well as for "Used cost-based optimizer" but this one is now always `yes`), so I think it is something for UI team to look at.

> In fact, why would that be the case?
In my scenario I explicitly changed `vectorize` setting manually. The users will probably not be doing that, so they will see either "no" or "yes". There is, however, a scenario when the user might see "no vectorization" for short period of time (while table statistics haven't been collected) which switches to "used vectorization" (because we rely on statistics to make a decision whether to vectorize or not, and if stats are not present, we do not vectorize).

> Is this an indication that we shouldn't actually be grouping these two queries under the same fingerprint?
I think the current grouping is reasonable, and I don't have a good feeling of what would be more preferable grouping criteria.

**Comment by awoods187** [comment]
Alright, I opened up https://github.com/cockroachdb/cockroach/issues/50284 to discuss this further--don't let that conversation impact this PR.

**Review by koorosh — APPROVED** [review]
LGTM (regarding UI changes)

File: pkg/sql/distsql_running.go [hunk 1]
```diff
@@ -261,10 +261,10 @@ func (dsp *DistSQLPlanner) setupFlows(
 	defer physicalplan.ReleaseSetupFlowRequest(&localReq)
 	ctx, flow, err := dsp.distSQLSrv.SetupLocalSyncFlow(ctx, evalCtx.Mon, &localReq, recv, localState)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
-	return ctx, flow, nil
+	return ctx, flow, setupReq.EvalContext.Vectorize != int32(sessiondata.VectorizeOff), nil
```
Comment (rohany) [inline pkg/sql/distsql_running.go:267]: why don't we check here if the flow was vectorized or not? Why do we look at a setting

**Review by rohany — COMMENTED** [review]
stats related changes look fine

**Review by yuzefovich — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50255)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RaduBerinde and @rohany)

___
*[pkg/sql/app_stats.go, line 146 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50255#-M9vS9pQJSCbaF9_3Ee2-r1-146:-MAWOaY-A0byyy6Crzr2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/86907fbcfbdbb2b8d57929e94b9897a317cbfe76/pkg/sql/app_stats.go#L146)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

[nit]: this is too long for one line now
</blockquote></details>

Done.
___
*[pkg/sql/distsql_running.go, line 267 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50255#-M9vS9pQJSCbaF9_3Ee3-r1-267:-MAWPHQlDBEBGVr67UwJ:b-sd7153) ([raw file](https://github.com/cockroachdb/cockroach/blob/86907fbcfbdbb2b8d57929e94b9897a317cbfe76/pkg/sql/distsql_running.go#L267)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

why don't we check here if the flow was vectorized or not? Why do we look at a setting
</blockquote></details>

Indeed, good point, fixed.

**Comment by yuzefovich** [comment]
TFTRs!

bors r+

**Pull request merged**